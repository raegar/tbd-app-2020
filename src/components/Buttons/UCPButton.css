.finalbutton{
  background-color: #005C6E;
  color: black;
  width: 100px;
  height: 50px;
  font-size: large;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  border-radius: 25px;
  border: 2px solid black;
}

.smallbutton{
  background-color: #31B7BC;
  color: black;
  width: 100px;
  height: 50px;
  font-family: 'Montserrat', sans-serif;
  font-size: 16px;    border-radius: 25px;
  border: 2px solid #005C6E;
  outline: none !important;
}

.mediumbutton{
  background-color: #31B7BC;
  color: black;
  width: 125px;
  height: 75px;
  font-family: 'Montserrat', sans-serif;
  font-size: 16px;    border-radius: 25px;
  border: 2px solid #005C6E;
  outline: none !important;
}

.largebutton{
    background-color: #31B7BC;
    color: black;
    width: 200px;
    height: 100px; 
    font-family: 'Montserrat', sans-serif;
    font-size: 22px;    border-radius: 25px;
    border: 2px solid #005C6E;
    outline: none !important;
}

.extralargebutton{
    background-color: #31B7BC;
    color: black;
    width: 400px;
    height: 200px;
    font-family: 'Montserrat', sans-serif;
    font-size: 22px;    border-radius: 25px;
    border: 2px solid #005C6E;
    outline: none !important;
}

.admin-navbar-button{
  background-color: #005C6E;
  color: #797979;
  font-family: 'Montserrat', sans-serif;
  font-size: 20px;
  border-radius: 0;
  border: 0;
  outline: none !important;
}

.admin-navbar-button:hover{
  background-color: #005C6E;
  color: #D5D5D5;
}

.admin-navbar-button-active{
  background-color: #fff;
  color: #797979;
  font-family: 'Montserrat', sans-serif;
  font-size: 20px;
  border-radius: 0;
  border: 0;
  outline: none !important;
}

  .smallbutton:hover{
    background-color: white;
    color: black;
    width: 100px;
    height: 50px;
    font-family: 'Montserrat', sans-serif;
    font-size: 16px;    border-radius: 25px;
    border: 2px solid #005C6E;
    outline: none !important;

  }

  .mediumbuttom:hover{
    background-color: white;
    color: black;
    width: 400px;
    height: 200px;
    font-family: 'Montserrat', sans-serif;
    font-size: 22px;    border-radius: 25px;
    border: 2px solid #005C6E;
    outline: none !important;

  }

  .largebutton:hover{
    background-color: white;
    color: black;
    width: 200px;
    height: 100px; 
    font-family: 'Montserrat', sans-serif;
    font-size: 22px;    border-radius: 25px;
    border: 2px solid #005C6E;
    outline: none !important;

  }
  .extralargebutton:hover{
    background-color: white;
    color: black;
    width: 400px;
    height: 200px;
    font-family: 'Montserrat', sans-serif;
    font-size: 22px;    border-radius: 25px;
    border: 2px solid #005C6E;
    outline: none !important;

  }
