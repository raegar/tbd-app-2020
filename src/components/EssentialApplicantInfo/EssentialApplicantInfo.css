.es-form-background {
    width: 100%;
    background-color: #B2DDDF;
    display: inline-block;
    padding: 20px;
    /*text-align: left;*/
    border: 1px solid black;
    margin-bottom: 10px;
}

.centered-buttons {
    text-align: center;
}

.es-form-container {
    padding-right: 0px  ;
    padding-left: 0px !important;
}

.form-title {
    text-align: center;
}
.ess-form-right {
    float: right;
    width: 50%;
    margin-top: 30px;
}
.ess-form-left {
    float:left;
    width: 50%;
    padding-left: 40px;
}

.mobiletextbox input {
    width: 250px;
    text-align: center;
}
.mobiletextbox {
    width: 200px;
}
.mobiletextbox label {
    margin-left: 5px;
}

@media only screen and (max-width: 768px) {
    .ess-form-left {
    width: 100%;
    float: left;
    padding-left: 0px;
    }
    .ess-form-right {
        width: 100%;
        float: left;
    }
}

.h3 {
    font-family: 'Montserrat', sans-serif;
}
