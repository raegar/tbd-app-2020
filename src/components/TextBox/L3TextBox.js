import React from 'react';
import './TextBox.css';
import PropTypes from 'prop-types';
import {isMobile} from 'react-device-detect';

class L3TextBox extends React.Component {
	constructor(props) {
		super(props);

		this.props = props;
		this.state = {
			emptyInput: this.props.clearInput,
			input: '',
		};
	}

    handleChange = (e) => {
    	if (this.state.emptyInput) {
    		this.setEmptyInput(false);
    	}
        
    	this.setInput(e.currentTarget.value);
    	console.log(this.state.input);  
    }

    setEmptyInput = (input) => {
    	this.setState({emptyInput: input});
    }

    setInput = (input) => {
    	this.setState({input: input});
    }

    clearInputField = (e) => {
    	this.setEmptyInput(!this.props.clearInput);
    }

    render = () => {
    	const className = isMobile ? "mobiletextbox" : "textbox";
	
    	return (
    		<div className={`${className ? "mobiletextbox" : "textbox"}`} id={this.props.wrapID}>
    			<label htmlFor={this.props.id}>{this.props.title}</label>
    			<input
    				name={this.props.name}
    				placeholder={this.props.placeholder}
    				id={this.props.id}
    				key={this.props.key}
    				maxLength={this.props.maxLength}
    				minLength={this.props.minLength}
    				size={this.props.size}
    				type={this.props.type}
    				min={this.props.min}
    				max={this.props.max}
    				value={this.state.emptyInput ? '': this.state.input}
    				disabled={this.props.attribute}
    				onChange={this.handleChange}
    				clearInput={this.clearInputField}
    			/>
    		</div>
    	);
    }
}

L3TextBox.propTypes = {
	backgroundColor: PropTypes.string,
	title: PropTypes.string,
	size: PropTypes.number,
	maxLength: PropTypes.number,
	minLength: PropTypes.number,
	placeholder: PropTypes.string,
	type: PropTypes.oneOf(['text', 'password', 'email', 'tel', 'number', 'date']),
	min: PropTypes.number,
	max: PropTypes.number,
	wrapID: PropTypes.string,
	attribute: PropTypes.string
};

L3TextBox.defaultProps = {
	backgroundColor: null,
	size: 20,
	maxLength: 30,
	minLength: 0,
	placeholder: '',
	type: "text",
	attribute: ''
};

/* Example Use

<L3TextBox 
    title="First Name" 
    placeholder="Enter your firstname" 
    minLength="5" 
    maxLength="30" 
    id="input1" 
    type="text" 
/>

*/

export default L3TextBox;
