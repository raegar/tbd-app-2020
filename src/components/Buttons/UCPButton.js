import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import './UCPButton.css';

function UCPButton({to, className, buttonText, ...props}) {
	if (to === undefined) {
		to = "/emptyLink";
	} else if (to === "none") {
		return(
			<button data-testid="ucpButton"
				type="button"
				className={className}
				{...props}>
				{buttonText}​
			</button>
		);
	}  
	
	return (
		<Link to={to}>
			<button data-testid="ucpButton"
				type="button"
				className={className}
				{...props}>
				{buttonText}​
			</button>
		</Link>
	);
}

UCPButton.propTypes = {
	/**
   * Is this the principal call to action on the page?
   */
	className:PropTypes.string,
	/**
   * Optional click handler
   */
	onClick:PropTypes.func,
};

UCPButton.defaultProps = {
	size:'UCPDefault',
	onClick:undefined
};

export default UCPButton;

/* Example

<UCPButton to="NAMEOFPAGE" primary=true className="Large" buttonText="Click Here"/> 

*/


