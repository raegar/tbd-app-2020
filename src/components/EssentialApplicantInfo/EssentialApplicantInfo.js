import React from "react";
import DropdownMenu from "../DropdownMenu/DropdownMenu";
import TextBox from "../TextBox/TextBox";
import RadioButton from "../RadioButton/RadioButton";
import "./EssentialApplicantInfo.css";
import { Container, Row, Col } from "react-bootstrap";
import UCPButton from "../Buttons/UCPButton";
import Modal from "../Modal/Modal";
import {isMobile} from 'react-device-detect';

const nationalityOptions = [
	{ label: "UK National", id: "UK", value: "UK" },
	{ label: "EU National", id: "EU", value: "EU" },
	{ label: "Other", id: "Other", value: "Other" },
];

const criminalOptions = [
	{ label: "Yes", id: "crimeYes", value: "yes" },
	{ label: "No", id: "crimeNo", value: "no" },
];

const supportOptions = [
	{ label: "Yes", id: "supportYes", value: "yes" },
	{ label: "No", id: "supportNo", value: "no" },
];

const time = [
	{value: 'morning', label: '09:00am - 12:00pm'},
	{value: 'afternoon', label: '12:00pm - 15:00pm'},
	{value: 'evening', label: '15:00pm - 17:00pm'}
];

const days = [
	{value: 'monday', label: 'Monday'},
	{value: 'tuesday', label: 'Tuesday'},
	{value: 'wednesday', label: 'Wednesday'},
	{value: 'thursday', label: 'Thursday'},
	{value: 'friday', label: 'Friday'}
];

class EssentialApplicantInfo extends React.Component {
	constructor(props) {
		super(props);
		this.props = props;

		this.state = {
			input: ""
		};
	}

  setInput = (input) => {
  	this.setState({input: input});
  }

  handleChange = (event) => {
  	console.log(event.target.value);
  	this.setInput(event.target.value);
  }

  saveSelectedData = () => {
  	//TODO: Fix to make more Reacty and not horrible bad raw HTML method
  	global.ApplicationFormData.name = document.getElementById("nameinput").value;
  	global.ApplicationFormData.email = document.getElementById("emailinput").value;
  	global.ApplicationFormData.telephone = document.getElementById("telinput").value;
  	global.ApplicationFormData.dob = document.getElementById("dobinput").value;
  	global.ApplicationFormData.ucasNumber = document.getElementById("ucasinput").value;
  	// get Nationality
  	if (document.getElementById("UK").checked) {
  		global.ApplicationFormData.Nationality = document.getElementById("UK").value;
  	}
  	if (document.getElementById("EU").checked) {
  		global.ApplicationFormData.Nationality = document.getElementById("EU").value;
  	}
  	if (document.getElementById("Other").checked) {
  		global.ApplicationFormData.Nationality = document.getElementById("Other").value;
  	}
  	// Criminal Conviction
  	if (document.getElementById("crimeYes").checked) {
  		global.ApplicationFormData.Convictions = document.getElementById("crimeYes").value;
  	}
  	if (document.getElementById("crimeNo").checked) {
  		global.ApplicationFormData.Convictions = document.getElementById("crimeNo").value;
  	}
  	// Learning Support
  	if (document.getElementById("supportYes").checked) {
  		global.ApplicationFormData.Support = document.getElementById("supportYes").value;
  	}
  	if (document.getElementById("supportNo").checked) {
  		global.ApplicationFormData.Support = document.getElementById("supportNo").value;
  	}
  	var course = document.getElementById("courseSelection");
  	global.ApplicationFormData.selectedCourse = course.options[course.selectedIndex].value;
  	//
  	console.log(global.ApplicationFormData);
  }

  userTypeSelect = () => {
  	if (global.userType === "staff") {
  		return (
  			<Row id="buttonRow">
  				<Col className="centered-buttons">
  					<UCPButton to="/Login"
  						primary="True"
  						className="mediumbutton"
  						buttonText="Go Back"
  					/>
  				</Col>
  				<Col className="centered-buttons">
  					<div id="confirmButton" onClick={this.saveSelectedData}>
  						<UCPButton to="/MoreInfo"
  							primary="True"
  							className="mediumbutton"
  							buttonText="Confirm and Go"
  						/>
  					</div>
  				</Col>
  			</Row>
  		);
  	} else {
  		return (
  			<Row id="buttonRow">
      
  				<Col className="centered-buttons">
  					<div id="confirmButton1">
  						<Modal id="hello" btnText="Request a Call Back">
  							<div>
  								<h2>Callback Request</h2>
  								<p>Please select a day and time you would be available.</p>
  								<Row id="dropdownRow">
  									<Col className="dropdownOptions1">
  										<DropdownMenu placeholder="Please select a day..." options={days}/>
  									</Col>
  									<Col className="dropdownOptions2">
  										<DropdownMenu placeholder="Please select a time..." options={time}/>
  									</Col>
  								</Row>
  								<br></br>
  								<UCPButton 
  									to="none"
  									primary="True"
  									className="smallbutton"
  									buttonText="Submit" />
  							</div>
  						</Modal>
  					</div>
            
  				</Col>
  				<Col className="centered-buttons">
  					<div id="confirmButton" onClick={this.saveSelectedData}>
  						<UCPButton
  							to="/MoreInfo"
  							primary="True"
  							className="mediumbutton"
  							buttonText="Apply Now"
  						/>
  					</div>
  				</Col>

  			</Row>
  		);
  	}
  }

  render = () => {
  	let textboxClassName;
  	if (isMobile) {
  		textboxClassName = "mobiletextbox";
  	} else {
  		textboxClassName = "desktoptextbox";
  	}

  	return (
  		<Container className="es-form-container">
  			<div className="es-form-background">
  				<h3 className="form-title">CLEARING APPLICATION</h3>
  				<span className="ess-form-left">
  					<TextBox
  						title="Name"
  						placeholder="Enter your name"
  						minLength={5}
  						maxLength={60}
  						id="nameinput"
  						type="text"
  						name="name"
  						attribute="disabled"
  						className={textboxClassName}
  						onChange={this.handleChange}
  						value={this.state.input}
  					/>
  
  					<TextBox
  						title="Email"
  						placeholder="Enter your email"
  						minLength={5}
  						maxLength={60}
  						id="emailinput"
  						type="email"
  						name="email"
  						attribute="disabled"
  						className={textboxClassName}
  						onChange={this.handleChange}
  						value={this.state.input}
  					/>
  
  					<TextBox
  						title="Tel No"
  						placeholder="Enter phone number"
  						minLength="5"
  						maxLength="12"
  						id="telinput"
  						type="tel"
  						name="telno"
  						attribute="disabled"
  						className={textboxClassName}
  						onChange={this.handleChange}
  						value={this.state.input}
  					/>
  
  					<TextBox
  						title="D/O/B"
  						placeholder="Enter date of birth"
  						id="dobinput"
  						type="date"
  						name="dob"
  						attribute="disabled"
  						className={textboxClassName}
  						onChange={this.handleChange}
  						value={this.state.input}
  					/>
  
  					<TextBox
  						title="UCAS ID"
  						placeholder="Enter UCAS ID"
  						id="ucasinput"
  						type="text"
  						name="ucas"
  						attribute="disabled"
  						className={textboxClassName}
  						onChange={this.handleChange}
  						value={this.state.input}
  
  					/>
  				</span>
  				<span className="ess-form-right">
  					<RadioButton
  						heading="Nationality"
  						options={nationalityOptions}
  						name="nationality"
  						attribute="disabled"
  						onClick={this.handleChange}
  					/>
  
  					<RadioButton
  						heading="Convictions"
  						options={criminalOptions}
  						name="criminal"
  						subtitle="Do you have any unspent criminal convictions?"
  						attribute="disabled"
  					/>
  
  					<RadioButton
  						heading="Learning Support"
  						options={supportOptions}
  						name="support"
  						subtitle="Do you have any additional learning support needs or disabilities?"
  						attribute="disabled"
  					/>
  					<h4>Preferred Course</h4>
  					<DropdownMenu id="courseSelection"/>
  				</span>
  			</div>
  			<div>{this.userTypeSelect()}</div>
  		</Container>
  	);
  }
}

export default EssentialApplicantInfo;
