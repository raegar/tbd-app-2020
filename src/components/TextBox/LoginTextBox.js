import React from 'react';
import './TextBox.css';
import PropTypes from 'prop-types';
import {isMobile} from 'react-device-detect';

function LoginTextBox(wrapID, id, title, name, onChange, placeholder, maxLength, minLength, size, type, min, max, attribute) {
	const className = isMobile ? "mobiletextbox" : "textbox";

	return (
		<div className={`${className ? "mobiletextbox" : "textbox"}`} id={wrapID}>
			<label htmlFor={id}>{title}</label>
			<input
				name={name}
				onChange={onChange}
				placeholder={placeholder}
				id={id}
				maxLength={maxLength}
				minLength={minLength}
				size={size}
				type={type}
				min={min}
				max={max}
				disabled={attribute}
			/>
		</div>
	);
}

LoginTextBox.propTypes = {
	backgroundColor: PropTypes.string,
	title: PropTypes.string,
	size: PropTypes.number,
	maxLength: PropTypes.number,
	minLength: PropTypes.number,
	placeholder: PropTypes.string,
	type: PropTypes.oneOf(['text', 'password', 'email', 'tel', 'number', 'date']),
	min: PropTypes.number,
	max: PropTypes.number,
	wrapID: PropTypes.string,
	attribute: PropTypes.string
};

LoginTextBox.defaultProps = {
	backgroundColor: null,
	size: 20,
	maxLength: 30,
	minLength: 0,
	placeholder: '',
	type: "text",
	attribute: ''
};

/* Example Use

<TextBox 
    title="First Name" 
    placeholder="Enter your firstname" 
    minLength="5" 
    maxLength="30" 
    id="input1" 
    type="text" 
/>

*/

export default LoginTextBox;
