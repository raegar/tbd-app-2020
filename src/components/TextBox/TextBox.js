import React from 'react';
import './TextBox.css';
import PropTypes from 'prop-types';
import {isMobile} from 'react-device-detect';

function TextBox({
	wrapID, 
	id, 
	title, 
	name, 
	onChange, 
	placeholder, 
	maxLength, 
	minLength, 
	size, 
	type, 
	min, 
	max, 
	attribute,
	multiline, 
	...props}) {
	const textBoxClassName = isMobile ? "mobiletextbox" : "textbox";
	/*TODO: can we cull a bunch of these via the use of {...props}? it's basically meant to
	be a copy of input so some poor soul could get points for adding a component anyways.*/
	const textBoxPropsObj = {
		name: name,
		onChange: onChange,
		placeholder: placeholder,
		id: id,
		maxLength: maxLength,
		minLength: minLength,
		size: size,
		type: type,
		min: min,
		max: max,
		disabled: attribute,
		...props
	};

	//<input> doesn't work with multiple lines 
	return (
		<div className={`${textBoxClassName ? "mobiletextbox" : "textbox"}`} id={wrapID}>
			<label htmlFor={id}>{title}</label>
			{(!multiline) ? 
				<input 
					{...textBoxPropsObj}
				/>
				: <textarea
					{...textBoxPropsObj}
				/>}
		</div>
	);
}

TextBox.propTypes = {
	backgroundColor: PropTypes.string,
	title: PropTypes.string,
	size: PropTypes.number,
	maxLength: PropTypes.number,
	minLength: PropTypes.number,
	placeholder: PropTypes.string,
	type: PropTypes.oneOf(['text', 'password', 'email', 'tel', 'number', 'date']),
	min: PropTypes.number,
	max: PropTypes.number,
	wrapID: PropTypes.string,
	attribute: PropTypes.string
};

TextBox.defaultProps = {
	backgroundColor: null,
	size: 20,
	maxLength: 30,
	minLength: 0,
	placeholder: '',
	type: "text",
	attribute: ''
};

/* Example Use

<TextBox 
    title="First Name" 
    placeholder="Enter your firstname" 
    minLength="5" 
    maxLength="30" 
    id="input1" 
    type="text" 
/>

*/

export default TextBox;
