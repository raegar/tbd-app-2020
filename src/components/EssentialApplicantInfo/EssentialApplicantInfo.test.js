import React from 'react';
import { render, fireEvent, screen, queryAllByRole } from '@testing-library/react';
import '@testing-library/jest-dom';
import EssentialApplicantInfo from './EssentialApplicantInfo';
import { BrowserRouter as Router } from 'react-router-dom';
import TextBox from '../TextBox/TextBox';
import UCPButton from "../Buttons/UCPButton";
//cleans up 
it("Renders Essential Applicant Info", () => {
    const { asFragment } = render(<Router><EssentialApplicantInfo /></Router>);
    expect(asFragment()).toMatchSnapshot();
});
// check whether all text boxes are present
it("Renders Name TextBox and Placeholder", () => {
    const {queryByPlaceholderText} = render(<TextBox
                                            type="text" title="Name" placeholder="Enter your name"/> )
    expect(queryByPlaceholderText("Enter your name")).toBeTruthy()   
});
it("Renders Email TextBox and Placeholder", () => {
    const {queryByPlaceholderText} = render(<TextBox
                                            type="text" title="Email" placeholder="Enter your email"/> )
    expect(queryByPlaceholderText("Enter your email")).toBeTruthy()   
});
it("Renders Phone Number TextBox and Placeholder", () => {
    const {queryByPlaceholderText} = render(<TextBox
                                            type="text" title="Tel No" placeholder="Enter phone number"/> )
    expect(queryByPlaceholderText("Enter phone number")).toBeTruthy()   
});
it("Renders Date Of Birth TextBox and Placeholder", () => {
    const {queryByPlaceholderText} = render(<TextBox
                                            type="text" title="D/O/B" placeholder="Enter date of birth"/> )
    expect(queryByPlaceholderText("Enter date of birth")).toBeTruthy()   
});
it("Renders UCAS TextBox and Placeholder", () => {
    const {queryByPlaceholderText} = render(<TextBox
                                            type="text" title="UCAS ID" placeholder="Enter UCAS ID"/> )
    expect(queryByPlaceholderText("Enter UCAS ID")).toBeTruthy()   
});
// check whether the Apply Now button is present
it("Renders Confirm and Go Button", () => {
    const {queryByTestId, queryByPlaceholderName} = render(<UCPButton to="none" className="mediumbutton" buttonText="Apply Now​"/>)
    expect(queryByTestId("ucpButton")).toBeTruthy()

});
// check whether the Request a Call Back button is present
it("Renders Request a Call Back Button", () => {
    const {queryByTestId, queryByPlaceholderName} = render(<UCPButton to="none" className="mediumbutton" buttonText=" Request a Call Back​"/>)
    expect(queryByTestId("ucpButton")).toBeTruthy();

});
// test whether the onchange function works properply
describe("Input values", () => {
    it("updates name on change", () => {
        const { queryByPlaceholderText } = render(<TextBox placeholder="Enter your name" />)

        const textInput = queryByPlaceholderText('Enter your name')

        fireEvent.change(textInput, { target: { value: "Ernasi" } })

        expect(textInput.value).toBe("Ernasi")
    });

    it("updates email on change", () => {
        const { queryByPlaceholderText } = render(<TextBox placeholder="Enter your email" />)

        const textInput = queryByPlaceholderText('Enter your email')

        fireEvent.change(textInput, { target: { value: "ernasi.pikouli@peterborough.ac.uk" } })

        expect(textInput.value).toBe("ernasi.pikouli@peterborough.ac.uk")
    });
});

