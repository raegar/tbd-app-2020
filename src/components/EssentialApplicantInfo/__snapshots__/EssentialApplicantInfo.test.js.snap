// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Essential Applicant Info 1`] = `
<DocumentFragment>
  <div>
    <header>
      <a
        href="/tbd-app-2020"
      >
        <img
          alt="UCP official logo"
          class="UCPLogoBlue"
          data-testid="ucpLogoHeader"
          src="UCPLogo.png"
        />
      </a>
    </header>
    <div
      class="es-form-container container"
    >
      <div
        class="es-form-background"
      >
        <h3
          class="form-title"
        >
          CLEARING APPLICATION
        </h3>
        <span
          class="ess-form-left"
        >
          <div
            class="mobiletextbox"
          >
            <label
              for="nameinput"
            >
              Name
            </label>
            <input
              class="desktoptextbox"
              id="nameinput"
              maxlength="60"
              minlength="5"
              name="name"
              placeholder="Enter your name"
              required=""
              size="20"
              type="text"
              value=""
            />
          </div>
          <div
            class="mobiletextbox"
          >
            <label
              for="address-1"
            >
              Address Line 1
            </label>
            <input
              id="address-1"
              maxlength="100"
              minlength="10"
              name="address_line_1"
              placeholder="Address Line 1"
              size="35"
              type="text"
            />
          </div>
          <div
            class="mobiletextbox"
          >
            <label
              for="address-2"
            >
              Address Line 2
            </label>
            <input
              id="address-2"
              maxlength="100"
              minlength="10"
              name="address_line_2"
              placeholder="Address Line 2"
              size="35"
              type="text"
            />
          </div>
          <div
            class="mobiletextbox"
          >
            <label
              for="town-city"
            >
              Town/City
            </label>
            <input
              id="town-city"
              maxlength="100"
              minlength="10"
              name="town_city"
              placeholder="Town/City"
              size="35"
              type="text"
            />
          </div>
          <div
            class="mobiletextbox"
          >
            <label
              for="county"
            >
              County
            </label>
            <input
              id="county"
              maxlength="30"
              minlength="3"
              name="county"
              placeholder="County"
              size="25"
              type="text"
            />
          </div>
          <div
            class="mobiletextbox"
          >
            <label
              for="postcode"
            >
              Postcode
            </label>
            <input
              id="postcode"
              maxlength="9"
              minlength="6"
              name="postcode"
              placeholder="Postcode"
              size="10"
              type="text"
            />
          </div>
          <div
            class="mobiletextbox"
          >
            <label
              for="emailinput"
            >
              Email
            </label>
            <input
              class="desktoptextbox"
              id="emailinput"
              maxlength="60"
              minlength="5"
              name="email"
              placeholder="Enter your email"
              required=""
              size="20"
              type="email"
              value=""
            />
          </div>
          <div
            class="mobiletextbox"
          >
            <label
              for="telinput"
            >
              Phone No
            </label>
            <input
              class="desktoptextbox"
              id="telinput"
              maxlength="12"
              minlength="5"
              name="telno"
              placeholder="Enter phone number"
              size="20"
              type="tel"
              value=""
            />
          </div>
          <div
            class="mobiletextbox"
          >
            <label
              for="dobinput"
            >
              D/O/B
            </label>
            <input
              class="desktoptextbox"
              id="dobinput"
              maxlength="30"
              minlength="0"
              name="dob"
              placeholder="Enter date of birth"
              size="20"
              type="date"
              value=""
            />
          </div>
          <div
            class="mobiletextbox"
          >
            <label
              for="ucasinput"
            >
              UCAS ID
            </label>
            <input
              class="desktoptextbox"
              id="ucasinput"
              maxlength="12"
              minlength="5"
              name="ucas"
              placeholder="Enter UCAS ID"
              size="20"
              type="text"
              value=""
            />
          </div>
        </span>
        <span
          class="ess-form-right"
        >
          <div
            class="radio-buttons"
          >
            <h4>
              Nationality
            </h4>
            <p />
            <div
              class="radio-group"
            >
              <input
                data-testid="UK"
                id="UK"
                name="nationality"
                type="radio"
                value="UK"
              />
              <label
                for="UK"
              >
                UK National
              </label>
            </div>
            <div
              class="radio-group"
            >
              <input
                data-testid="EU"
                id="EU"
                name="nationality"
                type="radio"
                value="EU"
              />
              <label
                for="EU"
              >
                EU National
              </label>
            </div>
            <div
              class="radio-group"
            >
              <input
                data-testid="Other"
                id="Other"
                name="nationality"
                type="radio"
                value="Other"
              />
              <label
                for="Other"
              >
                Other
              </label>
            </div>
          </div>
          <div
            class="radio-buttons"
          >
            <h4>
              Convictions
            </h4>
            <p>
              Do you have any unspent criminal convictions?
            </p>
            <div
              class="radio-group"
            >
              <input
                data-testid="crimeYes"
                id="crimeYes"
                name="criminal"
                type="radio"
                value="yes"
              />
              <label
                for="crimeYes"
              >
                Yes
              </label>
            </div>
            <div
              class="radio-group"
            >
              <input
                data-testid="crimeNo"
                id="crimeNo"
                name="criminal"
                type="radio"
                value="no"
              />
              <label
                for="crimeNo"
              >
                No
              </label>
            </div>
          </div>
          <div
            class="radio-buttons"
          >
            <h4>
              Learning Support
            </h4>
            <p>
              Do you have any additional learning support needs or disabilities?
            </p>
            <div
              class="radio-group"
            >
              <input
                data-testid="supportYes"
                id="supportYes"
                name="support"
                type="radio"
                value="yes"
              />
              <label
                for="supportYes"
              >
                Yes
              </label>
            </div>
            <div
              class="radio-group"
            >
              <input
                data-testid="supportNo"
                id="supportNo"
                name="support"
                type="radio"
                value="no"
              />
              <label
                for="supportNo"
              >
                No
              </label>
            </div>
          </div>
        </span>
      </div>
      <div>
        <div
          class="row"
          id="buttonRow"
        >
          <div
            class="centered-buttons col"
          >
            <div
              id="confirmButton1"
            >
              <button
                class="mediumbutton"
                data-testid="ucpButton"
                type="button"
              >
                Request a Call Back​
              </button>
            </div>
          </div>
          <div
            class="centered-buttons col"
          >
            <div
              id="confirmButton"
            >
              <a
                href="/MoreInfo"
              >
                <button
                  class="mediumbutton"
                  data-testid="ucpButton"
                  primary="True"
                  type="button"
                >
                  Apply Now​
                </button>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <footer
      class="ucpFooter"
    />
  </div>
</DocumentFragment>
`;
